pal/ios-osx/tlsio_appleios.c

```diff
@@ -409,6 +409,36 @@ static int tlsio_appleios_close_async(CONCRETE_IO_HANDLE tls_io, ON_IO_CLOSE_COM
     return result;
 }
 
+char* CFStringCopyUTF8String(CFStringRef aString) {
+    if (aString == NULL)
+    {
+        return NULL;
+    }
+
+    CFIndex length = CFStringGetLength(aString);
+    CFIndex maxSize = CFStringGetMaximumSizeForEncoding(length, kCFStringEncodingUTF8) + 1;
+    char *buffer = (char *)malloc(maxSize);
+    if (CFStringGetCString(aString, buffer, maxSize, kCFStringEncodingUTF8))
```

Comment:
>CFStringGetCString [](start = 8, length = 18)

can this handle NULL in case the malloc failed?

Reply:
not sure... added handling for this.